pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -336,6 +354,34 @@ func isAdminCredSecret(namespace, secretName string) bool {
 	return false
 }
 
+// hasResourceTags returns true if the AWS resourceTags are present in the Infrastructure resource
+func hasResourceTags(event client.Object) bool {
+	switch infra := event.(type) {
+	case *configv1.Infrastructure:
+		if infra != nil {
+			if infra.Spec.PlatformSpec.AWS != nil {
+				if len(infra.Spec.PlatformSpec.AWS.ResourceTags) != 0 {
+					return true
+				}
+			}
+		}
+	default:
+		return false
+	}
+	return false
+}
+
+// areTagsUpdated validates and returns a true if the resourceTags are updated in the new event
+func areTagsUpdated(oldEvent, newEvent client.Object) bool {
+	if !hasResourceTags(newEvent) {
+		return false
+	}
```

Comment:
If a user removes the tags later as part of the update, do we want to remove them from aws user resource as well?

Reply:
Not really required, the tags remain in the unmanaged state.